psi4/src/psi4/libmints/oeprop.cc

```diff
@@ -1997,6 +1998,44 @@ std::tuple<double, double> get_mbis_params(int atomic_num, int shell_num) {
     return params[atomic_num][shell_num];
 }
 
+// A Helper Method That Calculates the Atomic Volume using Atomic Electron Densities derived from Charge Partitioning
+SharedMatrix compute_atomic_volumes(const std::shared_ptr<DFTGrid>& grid, const std::vector<double>& rho_a_points, const std::vector<double>& distances, int num_atoms) {
```

Comment:
Have you thought about how to route calcs through oeprop to avoid recalcuation and class member data, as Jonathon mentioned? would it be suitable to have both mbis charges and volumes routed through one function, and volumes do extra steps at the end? and if both present, only runs once.

Reply:
I set it up this way so that future charge partitioning schemes that are implemented can call the same function, allowing for reusability. Nothing is recalculated or copied in this design, as constant references are passed into the function.